cacher/cacher.go

```diff
@@ -106,7 +177,22 @@ func GetFromURLOrDefaultCacheWithCallback(req *http.Request, cacheDir string, on
 	return nil
 }
 
-func getCache(clogger logger.Logger, filepath string, onData func([]byte) error, remoteErr error) error {
+func getCache(url *url.URL, clogger logger.Logger, filepath string, onData func([]byte) error, remoteErr error) error {
+	// These vars can't be const, because you can't take references to constant values ¯\_(ツ)_/¯
```

Comment:
You don't really need to reference anything since the scope is so small (a single function with ~10 lines). You could simplify the function to something like:

```go
func getCache(url *url.URL, clogger logger.Logger, filepath string, onData func([]byte) error, remoteErr error) error {
    if err := doGetCache(clogger, filepath, onData, remoteErr); err != nil {
        totalCacheRequests.WithLabelValues("miss", url.String()).Inc()
    } else {
        totalCacheRequests.WithLabelValues("hit", url.String()).Inc()
    }

    return err
}
```